File: apps/api/src/search/schemas.ts
import { CollectionCreateSchema } from 'typesense/lib/Typesense/Collections';

export const schemas: CollectionCreateSchema[] = [
  {
    name: 'locations',
    enable_nested_fields: true,
    fields: [
      { name: 'id', type: 'string' },
      { name: 'name', type: 'string', index: true },
      { name: 'address', type: 'auto', index: true, facet: true },
      { name: 'website', type: 'string', optional: true },
      { name: 'phone', type: 'string', optional: true },
      { name: 'priceLevel', type: 'int32', facet: true },
      { name: 'accessibilityLevel', type: 'int32', facet: true },
      { name: 'description', type: 'string', index: true },
      { name: 'media', type: 'object[]' },
      { name: 'tags', type: 'string[]', facet: true },
      { name: 'categoryId', type: 'int32', facet: true },
      { name: 'coordinates', type: 'geopoint', index: true },
      { name: 'createdAt', type: 'auto' },
      { name: 'updatedAt', type: 'auto' },
    ],
  },
];

File: api/internal/pkg/search/schemas.go
package search

import (
	tsapi "github.com/typesense/typesense-go/v2/typesense/api"
	"github.com/typesense/typesense-go/v2/typesense/api/pointer"
)

type CollectionName string

const (
	CollectionPois CollectionName = "pois"
)

var schemas = []*tsapi.CollectionSchema{
	{
		Name:               string(CollectionPois),
		EnableNestedFields: pointer.True(),
		Fields: []tsapi.Field{
			{
				Name: "name",
				Type: "string",
			},
			{
				Name: "location",
				Type: "geopoint",
			},
			{
				Name:  "poi",
				Type:  "object",
				Facet: pointer.True(),
				Index: pointer.True(),
			},
		},
	},
}
